api/control/v1alpha1/types.go

```diff
@@ -257,3 +257,53 @@ type BindingList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Binding `json:"items"`
 }
+
+// CustomTransform describes how to select and transform some objects
+// on their way from WDS to WEC, without regard to the WEC (i.e.,
+// not changes that are specific to the individual WEC).
+// The transformation specified here is in addition to, and follows,
+// whatever is built into KubeStellar for that object.
+//
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,shortName={ct},categories={all}
+type CustomTransform struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec CustomTransformSpec `json:"spec,omitempty"`
+}
+
+// CustomTransformSpec selects some objects and describes how to transform them.
+// The selected objects are those that match the `apiGroup` and `resource` fields.
```

Comment:
Do we want to add a comment about future extension that will add more than just remove ?   (here or near the Remove array

Reply:
I have been beaten up about including remarks about potential future development. Even in documentation. if anything, let's preserve such thoughts in the Issue, #1976 .